### ISSUE TITLE
Why only some structs wrapping hash implement debug ?

### ISSUE THREAD

**Issue by AureliaDolo** [massalabs/massa#2242]
Structs marked with an x implement debug, other derive it, is this the behavior we want  ? 

* [ ] Address
* [ ] Private key
* [ ] Public Key
* [ ] Signature
* [ ] Endorsement Id
* [x] Operation Id
* [x] Block id
* [ ] SC output event id
* [x] Node id
* [ ] Hash

**Comment by damip** [comment]
we want to implement it everywhere because the derived Debug prints in hexadecimal, which is inconsistent with our base58 representation

**Comment by AureliaDolo** [comment]
By the way when it is implemented, why the code is the same as display ?

**Comment by damip** [comment]
because it does the same thing: it outputs the bs58check

**Comment by AureliaDolo** [comment]
Ok so in my opinion, it would be less confusing to useone implementation to implement the other.

**Comment by yvan-sraka** [comment]
Yes! This is a call for more consistency around all these structs that are just wrappers (Newtype constructs) of Hash type :)

**Comment by damip** [comment]
Yes, let's make those structs consistent :)
